modular_skyrat/modules/ashwalkers/code/buildings/railroad.dm

```diff
@@ -112,6 +113,15 @@
 	. = ..()
 	atom_storage?.show_contents(user)
 
+/obj/vehicle/ridden/rail_cart/attackby(obj/item/attacking_item, mob/user, params)
+	if(istype(attacking_item, /obj/item/stack/ore/glass))
+		var/obj/item/stack/ore/glass/use_item = attacking_item
+		if(GetComponent(/datum/component/simple_farm) || !use_item.use(10))
+			return ..()
+		AddComponent(/datum/component/simple_farm, TRUE, TRUE, list(0, 16))
+		return
+	return ..()
```

Comment:
Can you make it so that adding glass here makes it so the cart can no longer be ridden, and could you make it so using a shovel on a full cart gives it back the ridable component, and removes the sand (giving it back to you), without giving you back the seed (I think that makes sense gameplay-wise? Not too inflexible on that one, just feel like it would be in line with the other ways to farm).